js/main.js

```diff
@@ -0,0 +1,27 @@
+var $avatarUrl = document.querySelector('#avatarUrl');
+var $profileImage = document.querySelector('.placeholder');
+var $save = document.querySelector('button');
+var $username = document.querySelector('#username');
+var $fullName = document.querySelector('#fullName');
+var $location = document.querySelector('#location');
+var $bio = document.querySelector('#bio');
+
+function updateProfileImage(event) {
+  $profileImage.setAttribute('src', event.target.value);
+}
+
+function formSubmitted(event) {
+  data.profile.username = $username.value;
+  data.profile.fullName = $fullName.value;
+  data.profile.location = $location.value;
+  data.profile.bio = $bio.value;
+  data.profile.avatarUrl = $avatarUrl.value;
+  window.addEventListener('beforeunload', function () {
```

Comment:
This should be outside of your event handler. In general, you don't want to apply event listeners inside of event handlers. By doing this, every time your form is submitted, another event listener is added.